internal/config/config.go

```diff
@@ -389,9 +392,19 @@ func peerFromCR(p metallbv1beta2.BGPPeer, passwordSecrets map[string]corev1.Secr
 	if p.Spec.ASN == p.Spec.MyASN && p.Spec.EBGPMultiHop {
 		return nil, errors.New("invalid ebgp-multihop parameter set for an ibgp peer")
 	}
-	ip := net.ParseIP(p.Spec.Address)
-	if ip == nil {
-		return nil, fmt.Errorf("invalid BGPPeer address %q", p.Spec.Address)
+	var ip net.IP
+	if p.Spec.Address != "" && p.Spec.Interface == "" {
```

Comment:
let's move this below the following two checks, which are more validation.
There you can remove the `p.Spec.Interface == ""` too as you already enforced that above.

Reply:
done